pkgs/development/tools/build-managers/ninja/setup-hook.sh

```diff
@@ -32,11 +28,12 @@ fi
 ninjaInstallPhase() {
     runHook preInstall
 
-    installTargets="${installTargets:-install}"
-
     # shellcheck disable=SC2086
-    local flagsArray=( $installTargets \
-        $ninjaFlags "${ninjaFlagsArray[@]}")
+    local flagsArray=(
+        $ninjaFlags "${ninjaFlagsArray[@]}"
+        $installFlags "${installFlagsArray[@]}"
```

Comment:
Ninja does not have any flags that would be useful with the install target. I'd revert the change to `glib/setup-hook.sh` and remove this line, as it may break other setup hooks too.

Reply:
It looks like no package is using `buildFlags` with ninja (and the only package using `ninjaFlags` is `otfcc`); it seems counterproductive to support `installFlags`. I'd rather remove `buildFlags` from ninja flags array, since it is not needed and it is documented as a `make` flags variable.

Reply:
My opinion here is that we should reuse as much of the generic builder as possible. There are definitely some uses for this (for instance with checkFlags below). I can change the doc to reflect that {build|install|check}Flags passes flags to whatever build system you are using - not just make. As meson & cmake replace autotools, I think it is important we make them just as poewrful.